src/main/java/org/example/c07/Abstraction/MyLinkedList.java

```diff
@@ -0,0 +1,103 @@
+package org.example.c07.Abstraction;
+/*
+It implements NodeList.
+    -  It has one field of type ListItem called root.
+    -  A constructor that takes a ListItem and initialises the field root with the newly passed in parameter.
+    -  And four methods:
+        -  getRoot(), getter for root.
+        -  addItem(), takes a ListItem and returns true if it was added successfully or false otherwise.
+        If the item is already present, it doesn't get added. Use compareTo() to place the item in its proper order.
+        -  removeItem(), takes a ListItem and returns true if it was removed successfully or false otherwise.
+        -  traverse(), takes the root as an argument and does not return anything.
+        If the root is null it prints out: The list is empty, otherwise print each value on a separate line.
+
+ */
+
+public class MyLinkedList implements NodeList {
+    ListItem root;
+
+    public MyLinkedList(ListItem root) {
+        this.root = root;
+    }
+
+    @Override
+    public ListItem getRoot() {
+        return null;
+    }
+
+    @Override
+    public boolean addItem(ListItem newItem) {
+        if (newItem != null) {
```

Comment:
Something does not look good with this if condition and if block. Do an extra check.
Pobably should be
```
if (this.root == null) {
            // the tree was empty, so our item becomes the head of the tree
            this.root = newItem;
            return true;
        }
``` 
like in SearchTree.

Reply:
done.

Reply:
Where?